cunumeric/module.py

```diff
@@ -5327,6 +5336,630 @@ def nancumsum(
+        an array, a conversion is attempted.
+
+    axis : None or int or tuple[int], optional
+        Axis or axes along which a sum is performed.  The default,
+        axis=None, will sum all of the elements of the input array.
+        If axis is negative it counts from the last to the first axis.
+
+        If axis is a tuple of ints, a sum is performed on all of the
+        axes specified in the tuple instead of a single axis or all
+        the axes as before.
+
+    dtype : data-type, optional
+        The type of the returned array and of the accumulator in which
+        the elements are summed.  The dtype of `a` is used by default
+        unless `a` has an integer dtype of less precision than the
+        default platform integer.  In that case, if `a` is signed then
+        the platform integer is used while if `a` is unsigned then an
+        unsigned integer of the same precision as the platform integer
+        is used.
+
+    out : ndarray, optional
+        Alternative output array in which to place the result. It must
+        have the same shape as the expected output, but the type of
+        the output values will be cast if necessary.
+
+    keepdims : bool, optional
+        If this is set to True, the axes which are reduced are left
+        in the result as dimensions with size one. With this option,
+        the result will broadcast correctly against the input array.
+
+    initial : scalar, optional
+        Starting value for the sum. See `~cunumeric.ufunc.reduce` for
+        details.
+
+    where : array_like[bool], optional
+        Elements to include in the sum. See `~cunumeric.ufunc.reduce` for
+        details.
+
+    Returns
+    -------
+    nansum : ndarray, see `dtype` parameter above.
+        A new array holding the result is returned unless out is
+        specified, in which case it is returned. The result has the
+        same size as a, and the same shape as a if axis is not None or
+        a is a 1-d array.
+
+    See Also
+    --------
+    numpy.nansum, numpy.isnan, numpy.isfinite
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+
+    # Note that np.nansum and np.sum allow complex datatypes
+    # so there are no "disallowed types" for this API
+
+    # use prod if the datatype is not floating point type
+    allowed_dtypes: list[np.dtype[Any]] = [
+        np.dtype(np.float16),
+        np.dtype(np.float32),
+        np.dtype(np.float64),
+        np.dtype(np.complex64),
+        np.dtype(np.complex128),
+    ]
+    if a.dtype in allowed_dtypes:
+        unary_reduction_code = UnaryRedCode.NANSUM
+    else:
+        unary_reduction_code = UnaryRedCode.SUM
+
+    index_array = a._perform_unary_reduction(
+        unary_reduction_code,
+        a,
+        axis=axis,
+        dtype=dtype,
+        out=out,
+        keepdims=keepdims,
+        initial=initial,
+        where=where,
+    )
+
+    return index_array
```

Comment:
no need to store a value here

```suggestion
    return a._perform_unary_reduction(
        unary_reduction_code,
        a,
        axis=axis,
        dtype=dtype,
        out=out,
        keepdims=keepdims,
        initial=initial,
        where=where,
    )
```